phac-nml/irida-next
# Bot accounts UI

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds in the UI for project bot accounts (listing, removing, adding).

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/9c75a8cf-f1c9-4d07-8efd-2dfee5b5e343)

![image](https://github.com/phac-nml/irida-next/assets/25466211/8432664c-6413-4d59-97fd-f712563fff79)

![image](https://github.com/phac-nml/irida-next/assets/25466211/3dfc608a-6932-46a2-bb60-51d78d97db63)

![image](https://github.com/phac-nml/irida-next/assets/25466211/80579553-1f5f-4c30-9a86-8e6c339774df)

![image](https://github.com/phac-nml/irida-next/assets/25466211/bd5bf5c9-2afb-471c-9f7f-069e1000288b)

![image](https://github.com/phac-nml/irida-next/assets/25466211/4793c627-4d0b-478c-bcef-4f0af768429b)

![image](https://github.com/phac-nml/irida-next/assets/25466211/33e89e98-34ee-4393-b7be-cdb9207c3537)


## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com` or any other user
2. Create a project
3. From the sidebar on the project show page, click `Settings` and then `Bot Accounts`
4. You should not see any bot accounts
5. Click the `New bot account` button
6. Fill in the token name, access level, option expiration date for token, and select atleast 1 scope
7. Click `Submit`
8. You should see a table with the bot account that was created and above the table a section which lists the personal access token
9. Go to the `Members` page for the project
10. You should see the bot account that was created listed as a `Bot`
11. Go back to the `Bot Accounts` page for the project
12. Click the ellipsis for the bot account and click `Remove`. Confirm that you would like to remove the bot account
13. The table should update and no bot accounts should be listed
14. Go back to the `Members` page for the project
15. Verify that the bot account has also been removed from the project members


## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
